step3/src/api/items/index.ts

```diff
@@ -0,0 +1,87 @@
+import express, { Request, response, Response, Router } from 'express';
+import { TodoItem } from '../../domain/TodoItem';
+import Repository from '../../repository';
+
+const ItemsRouter: Router = express.Router();
+const repository = new Repository();
+
+ItemsRouter.get('/items', (_, res: Response) => {
+  res.json(repository.get());
+});
+
+ItemsRouter.post('/items', (req: Request, res: Response) => {
+  const content: string | undefined = req.body?.content;
+  if (content === undefined) {
+    res.status(400).json({ success: false, error: 'content를 입력해주세요.' });
+    return;
+  }
+
+  const todoItems: TodoItem[] = repository.get();
+  const newItem: TodoItem = {
+    idx: todoItems?.length !== 0 ? todoItems[todoItems.length - 1]?.idx + 1 : 0,
+    content,
+    completed: false,
+    createdAt: new Date(),
+  };
+  repository.set([...todoItems, newItem]);
+
+  res.status(200).json({ success: true });
+});
+
+ItemsRouter.put('/items/:idx', (req: Request, res: Response) => {
+  const content: string | undefined = req.body?.content;
+  if (content === undefined) {
+    res.status(400).json({ success: false, error: 'content를 입력해주세요.' });
+    return;
+  }
+
+  const todoItems: TodoItem[] = repository.get();
+  const idx: number = +req?.params?.idx;
```

Comment:
number로 type을 명시했지만, 실제로는 optional 때문에 undefined도 할당될 수 있습니다.

그래서 다음과 같이 작성해보면 어떨가요?
```suggestion
  const idx: number = +req?.params?.idx ?? -1;
```